cache/memcached_client.go

```diff
@@ -175,18 +176,24 @@ func newMemcachedClient(
 	client memcachedClientBackend,
 	selector updatableServerSelector,
 	config MemcachedClientConfig,
-	reg prometheus.Registerer,
+	legacyRegister prometheus.Registerer,
 	name string,
 ) (*memcachedClient, error) {
+	reg := prometheus.WrapRegistererWith(
+		prometheus.Labels{labelCacheBackend: backendMemcached},
+		prometheus.WrapRegistererWithPrefix(cacheMetricNamePrefix, legacyRegister))
+	legacyRegister = prometheus.WrapRegistererWithPrefix(legacyMemcachedPrefix, legacyRegister)
```

Comment:
Ops. Some renaming was applied to the input param too. The input registerer is not legacy, so should just be named `reg`.

I'm suggesting something like this:

```suggestion
	reg prometheus.Registerer,
	name string,
) (*memcachedClient, error) {
	legacyRegister := prometheus.WrapRegistererWithPrefix(legacyMemcachedPrefix, reg)
	reg = prometheus.WrapRegistererWith(
		prometheus.Labels{labelCacheBackend: backendMemcached},
		prometheus.WrapRegistererWithPrefix(cacheMetricNamePrefix, reg))
```

Reply:
I will check and apply the suggestion once I got back in front of my laptop.